### PR TITLE
Added Request.sign() option to omit oauth_body_hash

### PR DESCRIPTION
**Pull request by bensonk**
I'm using oauth2 to consume Twitter's streaming API.  To make the library compatible with their API, the `oauth_body_hash` header must be omitted.  To allow that, without breaking any existing functionality, I added a parameter defaulted to `True` called `omit_body_hash`.  With this small change, my code to consume twitter's streaming API works.

### CODE REVIEW

**Comment by joestump** [comment]
@rickhanlonii this seems like something that's relevant to your work in #138 as well?

**Comment by joestump** [comment]
@bensonk thanks for the PR! @jaitaiwan and @rickhanlonii are probably going to chime in on this as well. We need tests on this before we merge, but feel free to wait on those for now.

**Comment by rickhanlonii** [comment]
It looks like [section 4.1.1](http://oauth.googlecode.com/svn/spec/ext/body_hash/1.0/oauth-bodyhash.html#parameter) specifies when to include the `oauth_body_hash`:

>## 4.1.1.  When to Include the Body Hash
>Not all requests should contain the oauth_body_hash parameter.

> - OAuth Consumers SHOULD NOT include an oauth_body_hash parameter when making Request Token or Access Token OAuth requests.
> - OAuth Consumers MUST NOT include an oauth_body_hash parameter on requests with form-encoded request bodies.
> - OAuth Consumers SHOULD include the oauth_body_hash parameter on all other requests.

So I think this change in general is necessary according to 4.1.1. 

The only question I have here is whether or not the default value should be `True`. It seems to me that it would be straightforward to pass `omit_body_hash=True` for only calls to Request Token or Access Token OAuth and default to `False` for all other requests. 

In other words: I think the default value should match the requirements of "all other requests" rather than the two special defined cases. I understand from the above discussion that the requirements of "all other requests" to the twitter API would require `omit_body_hash=True`, but supporting a particular third party API should not be a motivating factor in the design at hand.

**Comment by rickhanlonii** [comment]
Also, I think it is more clear for the parameter to be defined positively as in `include_body_hash=True`, rather than negatively as in `omit_body_hash=False` since the latter requires parsing a double negative to understand that the body hash will be included.

**Comment by joestump** [comment]
@rickhanlonii I concur all around. @bensonk if you could add tests I'm :+1: after @jaitaiwan has reviewed.

**Comment by jaitaiwan** [comment]
@bensonk @joestump adding this to my long list of things to review.

**Comment by jaitaiwan** [comment]
LGTM after tests are added. @bensonk are you able to provide the tests for this?
